### ISSUE TITLE
How to correctly parallelize within a workflow in Neon.Temporal

### ISSUE THREAD

**Issue by phxnsharp** [nforgeio/neonKUBE#1232]
I have a workflow as such:
```
   a
  / \
 b   c
 |   |
 d   e
  \ /
   f
```

Using NewActivityFutureStub I am able to correctly start b and c in parallel. However, then I must choose to call either b OR c's `GetAsync()` and cannot call the other until I have fully finished awaiting the result. Since I don't know if b or c will complete first, I have no way to correctly start d or e reliably. 

I suppose a workaround would be to group b/d and c/d into their own child workflows, but I'd rather not add all that complexity. Are there any tricks I'm unaware of to wait for b or c to finish?

**Comment by phxnsharp** [comment]
Closing this as a duplicate of https://github.com/nforgeio/neonKUBE/issues/1105
